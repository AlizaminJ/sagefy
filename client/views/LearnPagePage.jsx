import React from 'react'
import ReactMarkdown from 'react-markdown'
import { number, string, shape } from 'prop-types'
import { convertUuidToUuid58 as to58 } from 'uuid58'
import Layout from './components/Layout'
import Icon from './components/Icon'

export default function LearnPagePage({
  hash,
  progress,
  card: {
    entityId,
    name,
    data: { body },
  },
}) {
  return (
    <Layout
      hash={hash}
      page="LearnPagePage"
      title="Learn"
      description="-"
      canonical={`/page-cards/${to58(entityId)}`}
    >
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}

      <section>
        <h1>{name}</h1>
        <ReactMarkdown source={body} />
      </section>

      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </Layout>
  )
}

LearnPagePage.propTypes = {
  hash: string.isRequired,
  progress: number,
  card: shape({
    name: string.isRequired,
    data: shape({
      body: string.isRequired,
    }).isRequired,
  }).isRequired,
}

LearnPagePage.defaultProps = {
  progress: null,
}
