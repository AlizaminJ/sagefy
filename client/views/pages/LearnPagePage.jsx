import React from 'react'
import ReactMarkdown from 'react-markdown'
import { number, string, shape } from 'prop-types'
import Icon from '../components/Icon'

export default function LearnPagePage({ progress, name, data: { body } }) {
  return (
    <div className="LearnPagePage">
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}

      <section>
        <h1>{name}</h1>
        <ReactMarkdown source={body} />
      </section>

      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </div>
  )
}

LearnPagePage.propTypes = {
  progress: number,
  name: string.isRequired,
  data: shape({
    body: string.isRequired,
  }).isRequired,
}

LearnPagePage.defaultProps = {
  progress: null,
}
