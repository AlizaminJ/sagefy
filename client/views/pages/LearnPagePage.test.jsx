import LearnPagePage from './LearnPagePage'

describe('LearnPagePage', () => {
  it('should render LearnPagePage', () => {
    expect(LearnPagePage({ data: {} })).toMatchSnapshot()
  })
})
