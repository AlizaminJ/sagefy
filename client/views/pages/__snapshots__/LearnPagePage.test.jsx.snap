// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnPagePage should render LearnPagePage 1`] = `
<Layout hash={hash}
  description="-"
  page="LearnPagePage"
  title="Learn"
>
  <section>
    <h1 />
    <ReactMarkdown
      astPlugins={Array []}
      escapeHtml={true}
      plugins={Array []}
      rawSourcePos={false}
      renderers={Object {}}
      skipHtml={false}
      sourcePos={false}
      transformLinkUri={[Function]}
    />
  </section>
  <section>
    <form
      action="/next"
    >
      <button
        type="submit"
      >
        <Icon
          i="card"
          s="m"
        />
         Next Card
      </button>
    </form>
  </section>
</Layout>
`;
